Java/Java.sublime-syntax

```diff
@@ -285,69 +414,144 @@ contexts:
             - match: "(?=})"
               pop: true
             - include: code
-  object-types:
-    - match: '\b((?:[a-z]\w*\.)*[A-Z]+\w*)<'
+
+  object-types-inherited:
+    - match: '\b{{qualified_id}}\b(?=\s*<)'
+      scope: entity.other.inherited-class.java
       push:
-        - meta_scope: storage.type.generic.java
-        - match: '>|[^\w\s,\?<\[\].]'
-          pop: true
-        - include: object-types
-        - match: <
-          comment: This is just to support <>'s with no actual type prefix
-          push:
-            - meta_scope: storage.type.generic.java
-            - match: '>|[^\w\s,\[\]<]'
+        - match: '<'
+          scope: punctuation.definition.generic.begin.java
+          set:
+            - meta_scope: meta.generic.java
+            - match: '>'
+              scope: punctuation.definition.generic.end.java
               pop: true
```

Comment:
Oh, correction: it's already included on line 429.  I was puzzled when I attempted to add a failing test and it passed.  :-)